File: day:21/isSameTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def check(self, head, result):
        if(head is None):
            result.append(None)
            return;
        result.append(head.val);
        self.check(head.left, result);
        self.check(head.right, result);
        
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        #base cases
        if(p is q): return True
        r1 = [];
        r2 = [];
    
        self.check(p, r1)
        self.check(q, r2);
    
        return str(r1) == str(r2);

File: day:21/isSameTree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun check(head: TreeNode?, result: MutableList<Int?>) {
        if(head == null){
            result.add(null);
            return;
        }
        result.add(head.`val`);
        check(head.left, result);
        check(head.right, result);
    }
    fun isSameTree(p: TreeNode?, q: TreeNode?): Boolean {
        // base cases
        if(p == q) return true
        var r1:MutableList<Int?> = mutableListOf(-1);
        var r2:MutableList<Int?> = mutableListOf(-1);
    
        check(p, r1)
        check(q, r2);
    
        return r1.toString() == r2.toString();
    }
}
